.changeset/wet-berries-report.md

```diff
@@ -0,0 +1,5 @@
+---
+'@apollo/server': patch
+---
+
+Updated documentStore to use a schema hash rather than a random UUID for custom implementations
```

Comment:
Changelog should really reflect user impact rather than implementation details. How about something like:
```suggestion
Previously, when users provided their own `documentStore`, Apollo Server used a random prefix per schema in order to guarantee there was no shared state from one schema to the next. Now Apollo Server uses a hash of the schema, which enables the provided document store to be shared if you choose to do so.
```